saleor/discount/migrations/0049_migrate_sales_to_promotions.py

```diff
@@ -191,25 +191,27 @@ def get_rule_by_channel_sale(rules_info):
 def channel_listing_in_batches(qs):
     first_sale_id = 0
     while True:
-        batch_1 = qs.filter(sale_id__gt=first_sale_id)[:BATCH_SIZE]
+        batch_1 = qs.values("sale_id").filter(sale_id__gt=first_sale_id)[:BATCH_SIZE]
         if len(batch_1) == 0:
             break
-        last_sale_id = batch_1[len(batch_1) - 1].sale_id
+        last_sale_id = batch_1[len(batch_1) - 1]["sale_id"]
 
         # `batch_2` extends initial `batch_1` to include all records from
         # `SaleChannelListing` which refer to `last_sale_id`
-        batch_2 = qs.filter(sale_id__gt=first_sale_id, sale_id__lte=last_sale_id)
+        batch_2 = qs.values("pk", "sale_id").filter(
+            sale_id__gt=first_sale_id, sale_id__lte=last_sale_id
+        )
         pks = list(batch_2.values_list("pk", flat=True))
```

Comment:
You will still hit database two times here:
1. `SELECT pk, sale_id ...` in line 208
2. `SELECT pk ...`in line 204

Instead you can do:
`pks = [v["pk"] for v in batch_2]`